### PR TITLE
[AzureMonitorExporter] misc cleanup

### PR DESCRIPTION
**Pull request by TimothyMothra**
### Changes
- cleanup AAD example in Demo project
- add AAD instructions to Readme
- remove unnecessary `NoWarn` from csproj

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/README.md [hunk 1]
```diff
@@ -62,7 +62,17 @@ The following examples demonstrate how to add the `AzureMonitorExporter` to your
 
 ### Authenticate the client
 
-Exporter does not use authentication. 
+Azure Active Directory (AAD) authentication is an optional feature that can be used with the Azure Monitor Exporter.
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/README.md:65]: We added this on the options itself.
https://github.com/Azure/azure-sdk-for-net/pull/34555. Need to decide on which one would be preferred way. IMO we should only have one way (may required breaking change).

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/README.md:65]: @rajkumar-rangaraj

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/README.md:65]: good catch, I missed this change.
lets use this PR to fix the API.

We're going to need to mark the current method as `Obsolete` to direct users to the new way.

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/README.md:65]: I opened https://github.com/Azure/azure-sdk-for-net/pull/35011 to address this.
Had an offline discussion and we're not going to take the change.

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by vishweshbankwar — APPROVED** [review]

**Pull request merged**